### PR TITLE
add futuresAllMarkPrices

### PR DESCRIPTION
**Pull request by hp8wvvvgnj6asjm7**

### CODE REVIEW

File: src/websocket.js [hunk 1]
```diff
@@ -687,6 +687,31 @@ const user = (opts, variator) => (cb, transform) => {
...
+  symbol: x.s,
+  markPrice: x.p,
+  indexPrice: x.i,
+  settlePrice: x.P,
+  fundingRate: x.r,
+  nextFundingRate: x.T,
+}))
+
+const futuresAllMarkPrices = (payload, cb, transform = true) => {
+
```
Comment (balthazar) [inline src/websocket.js:702]: neat: remove this newline

File: src/websocket.js [hunk 2]
```diff
@@ -687,6 +687,31 @@ const user = (opts, variator) => (cb, transform) => {
...
+  indexPrice: x.i,
+  settlePrice: x.P,
+  fundingRate: x.r,
+  nextFundingRate: x.T,
+}))
+
+const futuresAllMarkPrices = (payload, cb, transform = true) => {
+  
+  const variant = payload.updateSpeed === '1s' ? '!markPrice@arr@1s' : '!markPrice@arr'
+
```
Comment (balthazar) [inline src/websocket.js:704]: that one too

**Review by balthazar — CHANGES_REQUESTED** [review]

**Comment by hp8wvvvgnj6asjm7** [comment]
ẙ̗õ̖͔̬̰͚̝̩̌̆̒̏͛̆ͥ̋ͅͅu̟̺̝͎͈̘̤͉̹̰͚͕͍̣̦̣̰̰ͦ͌̊ͣ̉ͨ̔̄̏̈́̑ͧ̋̀ͨͯͣ̚'̼̪͚̹͍̰̭͔̺̗̺̝̫̻͈͈̥͍̳̜̼̬̓̅ͤ̇̓ͦ͑ͬ̄͂͊̃͒͗̓̂ͦͥ̏̓͋͐ͩͣ̏̚ͅͅͅͅr̞̝̻̰̤̦̘̱͎̘̖̩̟̬̝̰̝͍̹̫̫̹̼̼̖̹̟̫̜͇̗͚̪̙͊͊̇ͤͫ̐͂̑̒͗̉̆̈̽̃ͭ͆̌͆ͦ̉̿ͭ̑̃̎ͨ̎̌ͯ͂ͫ͋̐ͥͅe͍̪̯͈̖͎̯̝̱̼̲̖̦͍̜͕̯̫̦͖̪̙̬̮̭̣̩̥͙̣̪̝̜̼̟̭̱͓͚̬̭͉ͫ͒ͫͥ̎ͭ͆ͤͮ͛̐̃̉ͤ̊̾̊̌͂ͨͭͫ͌͋̊̇͒͌ͥͧͭ̾̒ͪͪͫͬ̍ͥͤ̄ͣͤͅ ̱͚̟̳̥͔̤͈̝̞̳̤̲̲̫̰̥̳̟̫̞̰͓͉̻̰͙͈͙̪͍̟̭̼̱̝̬̱̥͓̫̝̳̫̘̣͓͔̣̠͂ͧ̿ͯͭ̌̊ͯ̐̑̇ͪ̂̊̈́̉̅̄͒̂̾̐̾ͣ̎͋͐ͫ͊̅̂̽ͧͯ̍̏̑͊̔́ͯ̎͋̇̇̎̀ͦ̈͊̚̚ͅͅẁ̗͎̝͍͎̯͖̗̦̩͇̭̙̣͓͈͍̪̩̜̜̜̞̪͕͚̘͕͚͈̱̯̼̪̮̝̙͖̞͎̮̺̮̯̩̦̘̜̤̲͎̺͇͍̦̻̲͔̬͈͚̔̓ͤͯ͌͋ͬ̍̋̿̎͋̓͐̊͒̃̊̎̇̈́̀͐ͯ̄̊̿͗̾̽̓̔́ͮ̆̓ͨͤ̾̃͑̐̀͂͗̒͊̉̓ͣ̽̓́̓ͤ͌͂ͨ̚̚ẻ͉̱͚͕̦̮̦̬̟̳͔͈͙̰̭̭͍̠̟̞͔͕̺̝̠̦̝̺̫̞̬̘̳̝͍͎̦̭̪̲̞͉̻̲̘̠͈̫̱̳̜ͥ̌͆̓̑̅̇̍̽̆̊ͥ́̅ͥ̊̿́̑ͪ͊̂͂̍̔̃̀͑̀ͧ̀ͧ͌ͬͧ͗ͨ͊ͯ͌͋ͭ̽͋̿ͯ̔̾ͬ͗́ͅl͇̩͉̼͓̳̤͕̜̥̠̝͚̩̝̟̣̥͓͔̩͚͚̟̖̯͇͙̻̙̻̥̫̼͇̮̦̗̮̺̝̾͊ͪ́̉̋͂̏ͬ̓̾ͩͧ̃̊ͨ̎̈̈̅ͬͤͣͮ̆̋͐̊̃ͩ̾̅̓̋̃̐̈́ͫ̓͌̊̆̚ͅͅc̩̱̤͚̤͖̩̜̼̼͍̤͉̞̖͙̮̟̙̫̻̼̦̪͙̥̥̤͎͇̦̯͚̞̯͗̓̈̅ͯ̔̈́ͤ̒ͫͪ̍̍̌͌̎ͯͬ͒̓ͩ̒̽ͣ̌̐̇ͮ̇̅̒̇͒͊̚o̥̝̙̰͙͙̫͚͈̪̟̠̙̤̣͕̻̙̭̜͚̝͙͕͈͔̓̂̅̑̔͊͑̆̆̊ͥ̃̌͊̔ͥ͒̇͋ͩ̅̊̽̍ͬ͛m͚̺͔̬̞̗̖̪̺͖͉̤̭̪̗̣̤̲̓ͬ̈͋̈ͭͪ̔̏̿̊̄ͯ̽̀ͭ͑̂e̺͚̼̺̝̫̤̙̠̓̓ͪ̽ͪ̀̾̋̚

File: src/websocket.js [hunk 3]
```diff
@@ -687,6 +687,29 @@ const user = (opts, variator) => (cb, transform) => {
...
+const futuresAllMarkPrices = (payload, cb, transform = true) => {  
+  const variant = payload.updateSpeed === '1s' ? '!markPrice@arr@1s' : '!markPrice@arr'
+
+  const w = openWebSocket(`${endpoints.futures}/${variant}`)
+
+  w.onmessage = msg => {
+    const arr = JSON.parse(msg.data)
+    cb(transform ? futuresAllMarkPricesTransform(arr) : arr)
+  }
+  return options => cache.close(1000, 'Close handle was called', { keepClosed: true, ...options })
```
Comment (balthazar) [inline src/websocket.js:710]: Add a newline on top of this line

File: src/websocket.js [hunk 4]
```diff
@@ -687,6 +687,29 @@ const user = (opts, variator) => (cb, transform) => {
...
+  symbol: x.s,
+  markPrice: x.p,
+  indexPrice: x.i,
+  settlePrice: x.P,
+  fundingRate: x.r,
+  nextFundingRate: x.T,
+}))
+
+const futuresAllMarkPrices = (payload, cb, transform = true) => {  
+  const variant = payload.updateSpeed === '1s' ? '!markPrice@arr@1s' : '!markPrice@arr'
```
Comment (balthazar) [inline src/websocket.js:702]: can't the update speed also be 3s? or is that the default one?

**Comment by itman85** [comment]
@hp8wvvvgnj6asjm7 please check request change to continue process this PR, we need it merge soon. thank you a lot

**Pull request merged**